### ISSUE TITLE
Validation check in cursor.find() function.

### ISSUE THREAD

**Issue by planetarynebula** [kaist-cp/helpdesk#105]
I am implementing `attemptGet` function in `cursor.find()`. I am confused at how to implement validation check. There are two validation check in the paper.
1. `((node.version^nodeV) & IgnoreGrow) != 0`
2. `chV != Unlinked && child == node.child(dir)`

I think first check is done by `readGuard`'s validate function. But, I am confused at checking `Unlinked`. It is quite different to *Routing Node* which has `None` value. `Unlinked` one is deleted semantically, but not yet unlinked yet. But, how to implement this? Firist, I thought it can be implemented using `tag` in `crossbeam_epoch::Shared`. Is it right? But, I have no idea to use `WriteGuard` to change `tag` since I can only change `tag` value in `crossbeam_epoch::Atomic` type. 

To sum up, I have two question.
1. Do we have to implement `Unlinked Node`? Or is it enough to implement this as `Routing Node`?
2. If we have to implement `Unlinked Node` which is different to `Routing Node`, how to implement this? Changing `tag` value is needed?

**Comment by planetarynebula** [comment]
I thought about it, and now I think `Unlinked Node` is unnecessary. Because `remove` function will directly unlink a node or change it to `Routing Node`. In the paper, `remove` function does not unlink the node and just tag it as `Unlinked` due to concurrent rebalancing process (It's my opinion...). But, we don't need to care about rebalancing, so that tagging is unnecessary and then checking `Unlinked Node` is also unnecessary.

**Comment by tomtomjhj** [comment]
I stumbled upon some cases where a double free may occur when unlink detection doesn't work. My initial thought was that it's possible to detect unlinks just by read guard validation and checking if current node's parent's child is consistent with current node but this didn't work and I ended up using tags to mark unlinked nodes. However I still believe tags are not necessary and such double free can be prevented by proper use of locks. Maybe I did something wrong with locks.
